### PR TITLE
feat: add total price to invoice generator

### PR DESCRIPTION
**Pull request by emiliarepo**
## Description

Adds an automatically calculated sum for each individual item in an invoice

image

### Before submitting the PR, please make sure you do the following

- [x] Related to #547 
- [x] Prefix your PR title with feat:, fix:, chore:, or docs:.
- [x] This message body should clearly illustrate what problems it solves.
- [x] Make sure the commit history is linear, up-to-date with main branch and does not contain any erroneous changes

### Formatting and linting

- [x] Format code with `pnpm format` and lint the project with `pnpm lint`

### CODE REVIEW

File: apps/web/src/components/invoice-generator/index.tsx [hunk 1]
```diff
@@ -40,13 +40,16 @@ function InputRow({
...
   multiple,
   required,
   step,
   min,
-  onBeforeInput: onInput,
+  onBeforeInput,
+  onInput,
   unit,
   maxLength,
+  readOnly,
```
Comment (MikaelSiidorow) [inline apps/web/src/components/invoice-generator/index.tsx:52]: Should we pass all these props directly to the input instead of mapping it 1-by-1 when nothing is changed about the props?

Like

```ts
function InputRow({
  id,
  ...restProps
}: GenericFieldProps) {
// ...

<Input
  id={htmlId}
  type={type ?? "text"}
  {...restProps}
  >
```

File: apps/web/src/components/invoice-generator/index.tsx [hunk 2]
```diff
@@ -239,6 +245,14 @@ function InvoiceItem({
 }) {
   const t = useScopedI18n("invoicegenerator");
 
+  const [quantity, setQuantity] = useState(0);
+  const [unitPrice, setUnitPrice] = useState(0);
+  const [totalPrice, setTotalPrice] = useState(0);
+
+  useEffect(() => {
```
Comment (MikaelSiidorow) [inline apps/web/src/components/invoice-generator/index.tsx:252]: This now causes unnecessary renders when it's a separate state and updated via `useEffect`. Instead we could probably be calculating it directly on the fly from `quantity * unitPrice`

File: apps/web/src/components/invoice-generator/index.tsx [hunk 3]
```diff
@@ -239,6 +245,14 @@ function InvoiceItem({
...
+    setTotalPrice(quantity * unitPrice);
+  }, [quantity, unitPrice]);
+
   return (
     <fieldset>
```
Comment (MikaelSiidorow) [inline apps/web/src/components/invoice-generator/index.tsx:257]: Not part of this PR, but I think this fieldset nesting is a bit suspicious, might be worth taking a look at some point

Comment (emiliarepo) [reply apps/web/src/components/invoice-generator/index.tsx:225]: i dont know im just a dog

**Comment by emiliarepo** [comment]
changed color to gray

image

**Review by MikaelSiidorow — APPROVED** [review]

**Review by joonatanaatos — APPROVED** [review]
